src/Microsoft.DotNet.Interactive.Formatting/Formatter.cs

```diff
@@ -530,9 +554,9 @@ internal static ITypeFormatter InferPreferredFormatter(Type actualType, string m
 
             // Last resort use preferred mimeType
 
-            var preferredMimeType = GetPreferredMimeTypeFor(actualType);
+            var preferredMimeType = GetPreferredMimeTypesFor(actualType);
 
-            return GetPreferredFormatterFor(actualType, preferredMimeType);
+            return GetPreferredFormatterFor(actualType, preferredMimeType.FirstOrDefault());
```

Comment:
Can we really assume the first one is the preferred one? Since these are intended to be registered from different extensions that have no knowledge of one another, we should probably make this method return multiple formatters just like we've changed the other methods to return multiple MIME types.

Reply:
i like the idea